modules/usage-reports/src/test/java/com/ibm/cloud/platform_services/usage_reports/v4/UsageReportsIT.java

```diff
@@ -0,0 +1,379 @@
+    try {
+      GetResourceGroupUsageOptions getResourceGroupUsageOptions = new GetResourceGroupUsageOptions.Builder()
+      .accountId(ACCOUNT_ID)
+      .resourceGroupId(RESOURCE_GROUP_ID)
+      .billingmonth(BILLING_MONTH)
+      .names(true)
+      .build();
+
+      Response<ResourceGroupUsage> response = service.getResourceGroupUsage(getResourceGroupUsageOptions).execute();
+      assertNotNull(response);
+      assertEquals(response.getStatusCode(), 200);
+
+      ResourceGroupUsage resourceGroupUsageResult = response.getResult();
+      assertNotNull(resourceGroupUsageResult);
+      log(String.format("GetResourceGroupUsage response:\n%s", resourceGroupUsageResult.toString()));
+
+      assertEquals(resourceGroupUsageResult.getAccountId(), ACCOUNT_ID);
+      assertEquals(resourceGroupUsageResult.getMonth(), BILLING_MONTH);
+      assertNotNull(resourceGroupUsageResult.getResources());
+      assertFalse(resourceGroupUsageResult.getResources().isEmpty());
+    } catch (ServiceResponseException e) {
+        fail(String.format("Service returned status code %d: %s\nError details: %s",
+          e.getStatusCode(), e.getMessage(), e.getDebuggingInfo()));
+    }
+  }
+
+  @Test
+  public void testGetOrgUsage() throws Exception {
+      try {
+          GetOrgUsageOptions getOrgUsageOptions = new GetOrgUsageOptions.Builder()
+                  .accountId(ACCOUNT_ID)
+                  .organizationId(ORG_ID)
+                  .billingmonth(BILLING_MONTH)
+                  .names(true)
+                  .build();
+
+          Response<OrgUsage> response = service.getOrgUsage(getOrgUsageOptions).execute();
+          assertNotNull(response);
+          assertEquals(response.getStatusCode(), 200);
+
+          OrgUsage orgUsageResult = response.getResult();
+          assertNotNull(orgUsageResult);
+          log(String.format("GetOrgUsage response:\n%s", orgUsageResult.toString()));
+
+          assertEquals(orgUsageResult.getAccountId(), ACCOUNT_ID);
+          assertEquals(orgUsageResult.getMonth(), BILLING_MONTH);
+          assertNotNull(orgUsageResult.getResources());
+          assertFalse(orgUsageResult.getResources().isEmpty());
+      } catch (ServiceResponseException e) {
+          fail(String.format("Service returned status code %d: %s\nError details: %s", e.getStatusCode(),
+                  e.getMessage(), e.getDebuggingInfo()));
+      }
+  }
+
+  @Test
+  public void testGetResourceUsageAccount() throws Exception {
+      try {
+          GetResourceUsageAccountOptions getResourceUsageAccountOptions = new GetResourceUsageAccountOptions.Builder()
+                  .accountId(ACCOUNT_ID)
+                  .billingmonth(BILLING_MONTH)
+                  .names(true)
+                  .limit(50L)
+                  .build();
+
+          List<InstanceUsage> results = new ArrayList<>();
+
+          String offset = null;
+          boolean moreResults = true;
+
+          while (moreResults) {
```

Comment:
I didn't know we had code to handle pagination.

Reply:
well, we do only because I wrote it :)